src/lib/telemetry/hooks.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react'
+
+import type { GenericTelemetryClient } from './client.js'
+import { useTelemetryContext } from './TelemetryProvider.js'
+
+export function useTelemetryClient(): GenericTelemetryClient {
+  const { client } = useTelemetryContext()
+  return client
+}
+
+export function useTelemetryOnMount(name: string): void {
+  const telemetry = useTelemetryClient()
+  useEffect(() => {
+    telemetry.track('component_mounted', {
+      component_name: name,
+      is_custom_element: globalThis.seamCustomElementNames?.length > 0,
```

Comment:
@seveibar This is reliable unless the page has loaded the element bundle AND is using react, which should never be an expected use-case.

Reply:
also possible/maybe easier to implement this as a component wrapper, since the signature of the wrapper could be `<T>(Element:T) => T`

Reply:
I think a wrapper makes more sense because it'll be easier to inspect props or even use a `useMeasure` system

Reply:
I tried the wrapper but ran into some additional complexity. I think the hooks are cleaner for now and we can introduce the wrapper if it helps us send more data.

Reply:
sg